crates/db_views/src/post_view.rs

```diff
@@ -2192,6 +2204,56 @@ mod tests {
     Ok(())
   }
 
+  #[test_context(Data)]
+  #[tokio::test]
+  #[serial]
+  async fn post_listings_hide_media(data: &mut Data) -> LemmyResult<()> {
+    let pool = &data.pool();
+    let pool = &mut pool.into();
+
+    // Ensure the `hide_media` user setting is set
+    let local_user_form = LocalUserUpdateForm {
+      hide_media: Some(true),
+      ..Default::default()
+    };
+    LocalUser::update(pool, data.local_user_view.local_user.id, &local_user_form).await?;
+    data.local_user_view.local_user.hide_media = true;
```

Comment:
Also check that the image post is returned if `local_user.hide_media` is false. Easiest way is to update the image post first and then query it, before updating the LocalUser.